src/pynwb/base.py

```diff
@@ -37,7 +37,12 @@ def __to_dict(self, arg):
         if arg is None:
             return dict()
         else:
-            return {i.name: i for i in arg}
+            return_dict = {}
+            for i in arg:
+                assert not i.name is None # If a container doesn't have a name, it gets lost!
```

Comment:
This hunk is a bug fix, for when users supply containers with no name.  This allows early warning, instead of quietly losing the component in the dict